docs/5.0/kubernetes-access.md

```diff
@@ -45,109 +67,91 @@ Let's take a closer look at the available Kubernetes settings:
 - `listen_addr` defines which network interface and port the Teleport proxy server
   should bind to. It defaults to port 3026 on all NICs.
 
-### Connecting the Teleport proxy to Kubernetes
+## Setup
+
+Connecting the Teleport proxy to Kubernetes
 
 There are two options for setting up Teleport to access Kubernetes:
 
-## Option 1: Deploy Inside Kubernetes as a pod
+### Option 1: Teleport as a  "gateway" to multiple K8s Clusters
 
-Deploy Teleport Proxy service as a Kubernetes pod inside the Kubernetes cluster
-you want the proxy to have access to.
+A single central Teleport Data Plane acting as "gateway". Multiple Kubernetes clusters
+connect to it over reverse tunnels.
+
+The root Teleport Cluster should be setup following our standard config, to make sure
+clients can connect you must make sure that an invite token is set for the `kube`
+service and proxy_addr has `kube_lisetn_addr` set.
 
 ```yaml
-# snippet from /etc/teleport.yaml on the Teleport proxy service:
+# Example Snippet for the Teleport Root Service
+#...
+auth_service:
+  enabled: "yes"
+  listen_addr: 0.0.0.0:3025
+  tokens:
+  - kube:866c6c114724a0fa4d4d73216afd99fb1a2d6bfde8e13a19
+#...
 proxy_service:
-    # create the 'kubernetes' section and set 'enabled' to 'yes':
-    kubernetes:
-        enabled: yes
+  public_addr: proxy.example.com:3080
+  kube_listen_addr: 0.0.0.0:3026
 ```
 
-If you're using Helm, we've a chart that you can use. Run these commands:
+To get quickly setup, we provide a Helm chart that'll connect to the above root cluster.
 
 ```bash
+# Add Teleport Helm Repo
 $ helm repo add teleport https://charts.releases.teleport.dev
-$ helm install teleport teleport/teleport
-```
-You will still need a correctly configured `values.yaml` file for this to work. See
-our [Helm Docs](https://github.com/gravitational/teleport/tree/master/examples/chart/teleport#introduction) for more information.
 
-![teleport-kubernetes-inside](img/teleport-k8s-pod.svg)
+# Installing the Helm Chart
+helm install teleport-agent . \
+  --namespace teleport \
+  --set proxyAddr=$PROXY_ENDPOINT \
+  --set authToken=$JOIN_TOKEN \
+  --set kubeClusterName=$KUBERNETES_CLUSTER_NAME
+```
 
-## Option 2: Deploy Outside of Kubernetes
+| Things to set | Description |
+|-|-|
+| `proxyAddr` | The Address of the Teleport Root Service |
+| `authToken` | A static `kube` invite token |
+| `kubeClusterName` | Kubernetes Cluster name (there is no easy way to automatically detect the name from the environment) |
 
-Deploy the Teleport proxy service outside of Kubernetes and update the Teleport
-Proxy configuration with Kubernetes credentials. Update the Teleport Proxy
-configuration with Kubernetes credentials.
+### Option 2: Proxy running inside a k8s cluster.
```

Comment:
One concern with these updated options: it's not clear that you can still use the new helm chart above with this setup in Option 2.

We should instead explain:
- auth+proxy can run anywhere (inside or outside of k8s), and proxy must have `kube_listen_addr`
- options for connecting k8s clusters:
  1. `kubernetes_service` in a pod (plus new helm chart docs from above)
  2. `kubernetes_service` elsewhere, with kubeconfig (plus link to https://github.com/gravitational/teleport/blob/master/examples/k8s-auth/get-kubeconfig.sh for building kubeconfigs)

Reply:
What's required / instustiriocns for setting up the helm chart? Can you provide the options for what flags are needed for the helm chart.   Also, do we setup ingress / TLS? How's that handled for running inside the k8s cluster?

Reply:
Instructions for the `kubernetes_service` helm chart: https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-kube-agent/README.md
For `kubernetes_service` we don't need ingress or TLS certs, it connects over a reverse tunnel to the proxy (like IoT nodes).

Reply:
Thanks Andrew, I've added a bit more clarity and reworked your comment.  I plan to come back and break out this doc and hope to expand more with the next iteration.